### PR TITLE
Added unit tests for coordinate

### PR DESCRIPTION
**Pull request by seshasaibehara**
- Added unit tests for coordinate.
- Removed rewrap namespace and moved site,coordinate,lattice,structure into casmutils::xtal namespace.

### CODE REVIEW

File: tests/unit/casmutils/coordinate.cpp [hunk 1]
```diff
@@ -1,25 +1,66 @@
...
-    void SetUp() override
-    {
+    void SetUp() override{
         Eigen::Vector3d raw_coord(0.1, 0.2, 0.3);
+        Eigen::Matrix3d lattice;
+        lattice << 0,0.5,0.5,
+                0.5,0,0.5,
+                0.5,0.5,0;
+        
         coord0_ptr.reset(new rewrap::Coordinate(raw_coord));
```
Comment (skk74) [inline tests/unit/casmutils/coordinate.cpp]: I think that we are going to phase out the ```rewrap``` namespace. Please use ```casmutils::xtal``` to make this phasing out easy.

File: tests/unit/casmutils/coordinate.cpp [hunk 2]
```diff
@@ -1,25 +1,66 @@
...
     std::unique_ptr<rewrap::Coordinate> coord0_ptr;
     std::unique_ptr<rewrap::Coordinate> coord1_ptr;
+    std::unique_ptr<rewrap::Lattice> lattice_ptr;
 };
 
 TEST_F(CoordinateTest, Construct) { EXPECT_EQ(coord0_ptr->cart(), coord1_ptr->cart()); }
 
+TEST_F(CoordinateTest, FracRetrieve){
+    rewrap::Lattice &lattice_rewrap = *lattice_ptr;
+    Eigen::Matrix3d lattice;
```
Comment (skk74) [inline tests/unit/casmutils/coordinate.cpp]: The use of ```lattice_ptr``` here allows access to the ```casmutils::xtal::Lattice``` constructed in the test frame. You seem to be recreating a matrix that you made in the test frame which is code duplication. Consider putting the raw ```Eigen::Matrix3d``` in the test frame. Also please use variable names that help discern type. i.e. lattice_matrix for ```Eigen::Matrix3d``` and lattice for ```casmutils::xtal::Lattice```

File: tests/unit/casmutils/coordinate.cpp [hunk 3]
```diff
@@ -1,25 +1,66 @@
...
+    Eigen::Vector3d frac_coords = lattice.inverse()*coord0_ptr->cart();
+  
+    rewrap::Coordinate coords_rewrap = rewrap::Coordinate::from_fractional(frac_coords,lattice_rewrap);     
+    EXPECT_TRUE(frac_coords.isApprox(coords_rewrap.frac(lattice_rewrap)));
+}
+
+TEST_F(CoordinateTest, BringWithIn){
+    Eigen::Matrix3d lattice;
+    lattice << 0,0.5,0.5,0.5,0,0.5,0.5,0.5,0;
+    Eigen::Vector3d frac_coords = lattice.inverse()*coord0_ptr->cart();
```
Comment (skk74) [inline tests/unit/casmutils/coordinate.cpp]: it seems ```frac_coords``` is used in multiple tests. Consider initializing in the test frame.

File: tests/unit/casmutils/coordinate.cpp (hunk 1 above)
Comment (seshasaibehara) [reply tests/unit/casmutils/coordinate.cpp]: Done

File: tests/unit/casmutils/coordinate.cpp (hunk 2 above)
Comment (seshasaibehara) [reply tests/unit/casmutils/coordinate.cpp]: Done

File: tests/unit/casmutils/coordinate.cpp (hunk 3 above)
Comment (seshasaibehara) [reply tests/unit/casmutils/coordinate.cpp]: Done

**Pull request merged**